src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -51,39 +53,37 @@ function UserPropsProvider({ children }: any): ReactElement {
     logout({ returnTo: returnUrl });
   };
 
-  async function loadUser() {
-    setContextLoaded(false);
-    try {
-      const res = await getAccountInfo(token);
-      if (res.status === 200) {
-        const resJson = await res.json();
-        setUser(resJson);
-      } else if (res.status === 303) {
-        // if 303 -> user doesn not exist in db
-        setUser(null);
-        if (typeof window !== 'undefined') {
-          router.push('/complete-signup', undefined, { shallow: true });
+  React.useEffect(() => {
+    async function loadUser() {
+      setContextLoaded(false);
+      try {
+        const res = await getAccountInfo(token, tenantID);
+        if (res.status === 200) {
+          const resJson = await res.json();
+          setUser(resJson);
+        } else if (res.status === 303) {
+          // if 303 -> user doesn not exist in db
+          setUser(null);
+          if (typeof window !== 'undefined') {
+            router.push('/complete-signup', undefined, { shallow: true });
+          }
+        } else if (res.status === 401) {
+          // in case of 401 - invalid token: signIn()
+          setUser(false);
+          setToken(null);
+          loginWithRedirect({
+            redirectUri: `${process.env.NEXTAUTH_URL}/login`,
+            ui_locales: localStorage.getItem('language') || 'en',
+          });
+        } else {
+          // any other error
         }
-      } else if (res.status === 401) {
-        // in case of 401 - invalid token: signIn()
-        setUser(false);
-        setToken(null);
-        loginWithRedirect({
-          redirectUri: `${process.env.NEXTAUTH_URL}/login`,
-          ui_locales: localStorage.getItem('language') || 'en',
-        });
-      } else {
-        // any other error
+      } catch (err) {
+        console.log(err);
       }
-    } catch (err) {
-      console.log(err);
     }
-    setContextLoaded(true);
-  }
-
-  React.useEffect(() => {
-    if (token) loadUser();
-  }, [token]);
+    if (token && router.isReady && tenantID) loadUser();
+  }, [token, router.isReady, tenantID]);
```

Comment:
There is a logical issue in this code. If no tenant parameter is provided in the URL, the `loadUser()` function will never be called. As a result the user login process gets stuck.